File: ESTRUCTURA DE DATOS/DEBERES/SEGUNDO_PARCIAL/DEBER2/PROYECTO_NRC1978/LISTA_SIMPLE_PYTHON/Nodo.cpp
#include "Nodo.h"

template <typename T>
Nodo<T>::Nodo(T dato)
{
    this->dato = dato;
    this->siguiente = nullptr;
}

template <typename T>
T& Nodo<T>::getDato()  // Devolver una referencia en lugar de una copia
{
    return dato;
}

template <typename T>
void Nodo<T>::setDato(T dato)
{
    this->dato = dato;
}

template <typename T>
void Nodo<T>::setSiguiente(Nodo<T> *siguiente)
{
    this->siguiente = siguiente;
}

template <typename T>
Nodo<T> *Nodo<T>::getSiguiente()
{
    return this->siguiente;
}

File: ESTRUCTURA DE DATOS/DEBERES/SEGUNDO_PARCIAL/DEBER2/PROYECTO_NRC1978/LISTA_SIMPLE_PYTHON/PYTHON/nodo.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.siguiente = None

    def get_dato(self):
        return self.dato

    def set_dato(self, dato):
        self.dato = dato

    def set_siguiente(self, siguiente):
        self.siguiente = siguiente

    def get_siguiente(self):
        return self.siguiente
